planner/core/logical_plan_builder.go

```diff
@@ -3226,9 +3226,42 @@ func (b *PlanBuilder) buildUpdate(ctx context.Context, update *ast.UpdateStmt) (
 		tblID2table[id], _ = b.is.TableByID(id)
 	}
 	updt.TblColPosInfos, err = buildColumns2Handle(updt.OutputNames(), tblID2Handle, tblID2table, true)
+	if err == nil {
+		err = checkUpdateList(b.ctx, tblID2table, updt)
```

Comment:
it seems we only check it in update above normal executor..so point above point-get maybe hit this bug? :sob:

Reply:
@lysu  mmm... Then I should fix this or you should fix it?

Reply:
let me see see- - 

I'd fixed in https://github.com/pingcap/tidb/pull/23491 during a refactor, but maybe it's not suitable to cherry-pick to 4.0 or 3.0